### PR TITLE
Parsing into HOGs

### PR DESCRIPTION
**Pull request by austinhpatton**
Included module to parse orthogroups into phylogenetic hierarchical orthogroups using reconciled gene family trees and speciesrax species tree.

### CODE REVIEW

File: workflows/phylorthology.nf
Comment (mertcelebi) [inline workflows/phylorthology.nf:493]: you probably don't need the `collect()`s here

File: workflows/phylorthology.nf
Comment (mertcelebi) [inline workflows/phylorthology.nf:499]: let's just remove this for now

File: workflows/phylorthology.nf
Comment (mertcelebi) [inline workflows/phylorthology.nf:294]: I was gonna ask about this: why are these hard-coded? Can you add a comment explaining what these are doing?

Comment (mertcelebi) [reply workflows/phylorthology.nf:294]: And should we parametrize this?

File: modules/local/orthofinder_phylohogs.nf
Comment (mertcelebi) [inline modules/local/orthofinder_phylohogs.nf:9]: See my PR, these will need to change!

File: modules/local/orthofinder_phylohogs.nf
Comment (mertcelebi) [inline modules/local/orthofinder_phylohogs.nf:17]: unnecessary extra line

File: modules/local/orthofinder_phylohogs.nf
Comment (mertcelebi) [inline modules/local/orthofinder_phylohogs.nf:26]: You include this with bunch of them, but majority don't use it. Are these needed? I didn't want to remove without knowing but can you confirm if this is needed or not?

File: modules/local/orthofinder_phylohogs.nf
Comment (mertcelebi) [inline modules/local/orthofinder_phylohogs.nf:29]: So, here's the plan: 
* No more file HACKS!
* We'll first merge my S3 PR, and you'll have to pull those changes and adjust this process to work like I made my processes work (without relying on the results directory)

File: modules/local/orthofinder_phylohogs.nf
Comment (mertcelebi) [inline modules/local/orthofinder_phylohogs.nf:60]: None of this will be needed.

**Review by mertcelebi — CHANGES_REQUESTED** [review]
Overall fine, but let's make sure these are merged after my S3 PR is merged. So you can learn how orthofinder can work without relying on the results directory.

File: modules/local/generax.nf
Comment (mertcelebi) [inline modules/local/generax.nf:26]: You can do `path "GeneRax/results/**_reconciled_gft.newick", emit: generax_gfts` for both this and the SpeciesRax one

File: modules/local/generax.nf
Comment (mertcelebi) [inline modules/local/generax.nf:45]: I bet there's a non-loop version of this but I am not sure, so I'll let this slide!

File: modules/local/orthofinder_phylohogs.nf
Comment (mertcelebi) [inline modules/local/orthofinder_phylohogs.nf:21]: You need all of these?

**Review by mertcelebi — CHANGES_REQUESTED** [review]
Overall looks good. I left some comments/questions to address.

File: modules/local/generax.nf
Comment (austinhpatton) [reply modules/local/generax.nf:45]: I suspect so too, just gets tricky since they're all in their own directories, each of which is created from the single call of the generax program.

File: modules/local/orthofinder_phylohogs.nf
Comment (austinhpatton) [reply modules/local/orthofinder_phylohogs.nf:35]: Added in an expanded comment section describing the motivation behind this prep-work and have renamed the two variables (results_dir & dir) to "of_results_dir" and "of_working_dir" respectively ("of" being orthofinder - outlined in a comment above the variable names).

File: modules/local/orthofinder_phylohogs.nf
Comment (austinhpatton) [reply modules/local/orthofinder_phylohogs.nf]: You and me both! Just aggravating that it's necessary.

File: workflows/phylorthology.nf
Comment (austinhpatton) [reply workflows/phylorthology.nf]: Yeah, these should definitely be parameterized, as should a number of other things. 

I think that I'll make a list of what all would ideally be parameterized (across all modules) and then I'll create an issue listing these. Then, we can do a new PR focused on this?

File: modules/local/orthofinder_phylohogs.nf
Comment (austinhpatton) [reply modules/local/orthofinder_phylohogs.nf:30]: I'm not confident, but I suspect this has something to do with an option where you can specify additional arguments/parameters in the modules.json file. So, this is like one of three ways to specify user-input parameters that are not hard-coded as parameter options with default values?

Comment (austinhpatton) [reply modules/local/orthofinder_phylohogs.nf:30]: Regardless, I think it's only necessary/useful for some - this one not being one, so I'll get rid of here.

File: modules/local/orthofinder_phylohogs.nf
Comment (austinhpatton) [reply modules/local/orthofinder_phylohogs.nf:21]: Sadly I do. Doesn't actively use them all, but it looks to make sure they're all there, and if not it has a fit and won't run.

Comment (mertcelebi) [reply modules/local/orthofinder_phylohogs.nf:21]: Smh OF

File: workflows/phylorthology.nf
Comment (mertcelebi) [inline workflows/phylorthology.nf:103]: Boop

File: workflows/phylorthology.nf
Comment (mertcelebi) [reply workflows/phylorthology.nf]: Let's create an issue for this @austinhpatton

**Review by mertcelebi — APPROVED** [review]
These changes look good. Approving with two changes requested:
* Fix the line I commented "boop"
* Create an issue for the things to parameterize later

**Pull request merged**
